dataprofiler/profilers/profile_builder.py

```diff
@@ -580,13 +591,23 @@ def _update_row_statistics(self, data):
 
         :param data: a dataset
         :type data: pandas.DataFrame
+        :param sample_ids: list of indices in order they were sampled in data
+        :type sample_ids: list(int)
         """
+
+        if not isinstance(data, pd.DataFrame):
+            raise ValueError("Cannot calculate row statistics on data that is"
+                             "not a DataFrame")
         
         self.total_samples += len(data)
         self.hashed_row_dict = dict.fromkeys(
             pd.util.hash_pandas_object(data, index=False), True
         )
 
+        # Calculate complete rows read
+        # Only consider nulls in fully sampled rows for calculations
+        complete_row_ids = sample_ids[:self._min_col_samples_used]
```

Comment:
Probably should check if `sample_ids` exists

Reply:
if it doesn't exist, then what do we do?